yjit.rb

```diff
@@ -202,9 +202,16 @@ def self.disasm(iseq) # :nodoc:
     iseq = RubyVM::InstructionSequence.of(iseq)
 
     if self.enabled?
-      # Produce the disassembly string
-      # Include the YARV iseq disasm in the string for additional context
-      iseq.disasm + "\n" + Primitive.rb_yjit_disasm_iseq(iseq)
+      disasm_str = Primitive.rb_yjit_disasm_iseq(iseq)
+
+      if !disasm_str
+        $stderr.puts("YJIT disasm is only available when YJIT is built in dev mode, i.e.:\n")
+        $stderr.puts("./configure --enable-yjit=dev (see doc/yjit/yjit.md)\n")
+      else
+        # Produce the disassembly string
+        # Include the YARV iseq disasm in the string for additional context
+        iseq.disasm + "\n" + disasm_str
+      end
```

Comment:
If enabled but not a dev build this will return nil.
If not enabled it returns `iseq.disaam`.

Would it be better to consistently return `iseq.disasm` and just warn if we can't add the yjit disasm to it?  Or do you see value in returning the `nil` here?

Also, is there a reason to (or not to) use `warn` here?

Reply:
I didn't know about `warn`. I'm assuming this also prints to standard error?

I think returning `nil` might be useful because otherwise errors like this can go unnoticed, but maybe we should always return `nil` instead.

Reply:
Returning `nil` more often might be the least confusing route

Reply:
Yes, warn prints to stderr and includes stacktrace. Warnings can be also disabled
https://docs.ruby-lang.org/en/3.3/Kernel.html#method-i-warn